### PR TITLE
[core] Support timestamp field type in record level expire

### PR DESCRIPTION
**Pull request by askwang**
### Purpose

Record level expiration supports timestamp type and timestamp_ltz type.

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/io/RecordLevelExpire.java [hunk 1]
```diff
@@ -62,27 +66,39 @@ public static RecordLevelExpire create(CoreOptions options, RowType rowType) {
...
-        if (!((timeFieldType == CoreOptions.TimeFieldType.SECONDS_INT
-                        && field.type() instanceof IntType)
-                || (timeFieldType == CoreOptions.TimeFieldType.SECONDS_LONG
-                        && field.type() instanceof BigIntType)
-                || (timeFieldType == CoreOptions.TimeFieldType.MILLIS_LONG
-                        && field.type() instanceof BigIntType))) {
+        if (!isValidateFieldType(timeFieldType, field)) {
             throw new IllegalArgumentException(
                     String.format(
                             "The record level time field type should be one of SECONDS_INT,SECONDS_LONG or MILLIS_LONG, "
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/io/RecordLevelExpire.java:72]: The record level time field type should be one of SECONDS_INT, SECONDS_LONG, MILLIS_LONG or TIMESTAMP, but time field type is %s, field type is %s.

File: paimon-core/src/main/java/org/apache/paimon/io/RecordLevelExpire.java [hunk 2]
```diff
@@ -62,27 +66,39 @@ public static RecordLevelExpire create(CoreOptions options, RowType rowType) {
...
         }
 
-        return new RecordLevelExpire(fieldIndex, (int) expireTime.getSeconds(), timeFieldType);
+        return new RecordLevelExpire(
+                fieldIndex, (int) expireTime.getSeconds(), timeFieldType, field);
+    }
+
+    private static boolean isValidateFieldType(
+            CoreOptions.TimeFieldType timeFieldType, DataField field) {
+        return ((timeFieldType == CoreOptions.TimeFieldType.SECONDS_INT
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/io/RecordLevelExpire.java:83]: DataType dataType = field.type();

File: paimon-core/src/test/java/org/apache/paimon/table/RecordLevelExpireWithTimestampTest.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        options.set(CoreOptions.RECORD_LEVEL_TIME_FIELD_TYPE, CoreOptions.TimeFieldType.TIMESTAMP);
+        return options;
+    }
+
+    @Test
+    public void test() throws Exception {
+        long millis = System.currentTimeMillis();
+        Timestamp timestamp1 = Timestamp.fromEpochMillis(millis - 60 * 1000);
+        Timestamp timestamp2 = Timestamp.fromEpochMillis(millis);
+        Timestamp timestamp3 = Timestamp.fromEpochMillis(millis + 60 * 1000);
```
Comment (wwj6591812) [inline paimon-core/src/test/java/org/apache/paimon/table/RecordLevelExpireWithTimestampTest.java:83]: Add a ut to test LocalZonedTimestampType?

File: paimon-core/src/test/java/org/apache/paimon/table/RecordLevelExpireWithTimestampTest.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        Options options = new Options();
+        options.set(CoreOptions.BUCKET, 1);
+        options.set(CoreOptions.RECORD_LEVEL_EXPIRE_TIME, Duration.ofSeconds(1));
+        options.set(CoreOptions.RECORD_LEVEL_TIME_FIELD, "col1");
+        options.set(CoreOptions.RECORD_LEVEL_TIME_FIELD_TYPE, CoreOptions.TimeFieldType.TIMESTAMP);
+        return options;
+    }
+
+    @Test
+    public void test() throws Exception {
```
Comment (wwj6591812) [inline paimon-core/src/test/java/org/apache/paimon/table/RecordLevelExpireWithTimestampTest.java:79]: test what?

**Comment by wwj6591812** [comment]
Please add Purpose of this pr.

**Comment by askwang** [comment]
> Please add Purpose of this pr.

addressed, thanks for your review @wwj6591812

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java [hunk 5]
```diff
@@ -1351,7 +1351,7 @@ public class CoreOptions implements Serializable {
                     .enumType(TimeFieldType.class)
                     .defaultValue(TimeFieldType.SECONDS_INT)
```
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/CoreOptions.java:1352]: Can we just set this to `AUTO`? If the field is timestamp, we don't need to set this option.

Comment (askwang) [reply paimon-common/src/main/java/org/apache/paimon/CoreOptions.java:1352]: `AUTO` maybe cannot adapt to all types, just added detailed log.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**